From: Alexander Wolf <alex.v.wolf@gmail.com>
Date: Sun, 29 Jun 2014 18:03:50 +0000
Subject: Avoid using C isnan when we can use std::isnan REVIEW: 119010
X-Git-Tag: v4.13.80
X-Git-Url: http://quickgit.kde.org/?p=kstars.git&a=commitdiff&h=d3c530355f59991e7cc0b0a39485739802ea938b
---
Avoid using C isnan when we can use std::isnan REVIEW: 119010
---


--- a/kstars/skycomponents/asteroidscomponent.cpp
+++ b/kstars/skycomponents/asteroidscomponent.cpp
@@ -27,6 +27,8 @@
 #include "skyobjects/ksasteroid.h"
 #include "kstarsdata.h"
 #include "ksfilereader.h"
+
+#include <cmath>
 #include <kdebug.h>
 #include <kglobal.h>
 #include <kio/job.h>
@@ -194,7 +196,7 @@
         // FIXME: God help us!
         KSAsteroid *ast = (KSAsteroid*) so;
 
-        if ( ast->mag() > Options::magLimitAsteroid() || isnan(ast->mag()) != 0)
+        if ( ast->mag() > Options::magLimitAsteroid() || std::isnan(ast->mag()) != 0)
             continue;
 
         bool drawn = skyp->drawPointSource(ast,ast->mag());

--- a/kstars/skycomponents/cometscomponent.cpp
+++ b/kstars/skycomponents/cometscomponent.cpp
@@ -27,6 +27,8 @@
 #include "skylabeler.h"
 #include "skypainter.h"
 #include "projections/projector.h"
+
+#include <cmath>
 #include <kio/job.h>
 #include <kio/netaccess.h>
 #include <kio/jobuidelegate.h>
@@ -191,7 +193,7 @@
     foreach ( SkyObject *so, m_ObjectList ) {
         KSComet *com = (KSComet*)so;
         double mag= com->mag();
-        if (isnan(mag) == 0)
+        if (std::isnan(mag) == 0)
         {
             bool drawn = skyp->drawPointSource(com,mag);
             if ( drawn && !(hideLabels || com->rsun() >= rsunLabelLimit) )

--- a/kstars/skyobjects/starobject.cpp
+++ b/kstars/skyobjects/starobject.cpp
@@ -17,6 +17,7 @@
 
 #include "starobject.h"
 
+#include <cmath>
 #include <QColor>
 #include <QPainter>
 #include <QFontMetricsF>
@@ -282,7 +283,7 @@
 
     pmms = pmMagnitudeSquared();
 
-    if( isnan( pmms ) || pmms * num->julianMillenia() * num->julianMillenia() < 1. ) {
+    if( std::isnan( pmms ) || pmms * num->julianMillenia() * num->julianMillenia() < 1. ) {
         // Ignore corrections
         *ra = ra0().Degrees();
         *dec = dec0().Degrees();

